src/argparse_utils.h

```diff
@@ -0,0 +1,87 @@
+#include <boost/property_tree/ptree.hpp>
+#include <cxxopts.hpp>
+
+#include <valhalla/baldr/rapidjson_utils.h>
+#include <valhalla/config.h>
+#include <valhalla/filesystem.h>
+#include <valhalla/midgard/logging.h>
+#include <valhalla/midgard/util.h>
+
+// returns a (const char*)'s path's filename without the file extension, e.g.
+// /home/valhalla/valhalla_add_elevation.cc -> valhalla_add_elevation
+std::string file_stem(const char* path) {
+  auto basename = filesystem::path(path).filename().string();
+  if (basename.length() <= 3) {
+    return "";
+  }
+  basename.erase(basename.length() - 3);
```

Comment:
nit: are you sure you dont want to `rfind` `'.'`, i know its pedantic but its robust to us one day changing from stupid `.cc` for to the more prevalent `.cpp/.hpp`